protos/idl/service.proto

```diff
@@ -65,8 +65,9 @@ message Dataset {
 message DatasetID {
     string project = 1;  // The name of the project
     string name = 2;     // The name of the dataset
-    string domain = 3;   // The domain (eg. environment) in which it's desired to run (optional)
+    string domain = 3;   // The domain (eg. environment)
     string version = 4;  // Version of the data schema
+    string UUID = 5;     // UUID for the dataset
```

Comment:
Are these still all required fields from the perspective of the proto?  Will the datacatalog populate the project/name/domain/version fields for clients as long as they provide the UUID?

Reply:
In the cases of retrieving a Dataset/Artifact it should be optional.

In the cases of creating a Dataset, I wanted to ask you @milohan-lyft if there are cases where the client would like to specify the UUID (in the case where you want to use the UUID generated from another service/database).

Reply:
I don't think having the ability to specify a UUID will be a requirement from our end.

Reply:
Ok I will make this more clear in the proto in the next PR.